### PR TITLE
Add read_channel_content permission

### PR DESCRIPTION
**Pull request by enahum**
#### Ticket Link
https://mattermost.atlassian.net/browse/MM-53203

#### Release Note

```release-note
Added permission to read the contents of a channel
```

### CODE REVIEW

File: server/channels/app/permissions_migrations.go [hunk 1]
```diff
@@ -1092,6 +1092,25 @@ func (a *App) getProductsBoardsPermissions() (permissionsMap, error) {
...
+	}
+
+	t = append(t, permissionTransformation{
+		On: permissionOr(
+			isExactRole(model.SystemAdminRoleId),
+			isExactRole(model.ChannelUserRoleId),
+			isExactRole(model.ChannelGuestRoleId),
+		),
+		Add: readChannelContentPermissions,
+	})
```
Comment (lieut-data) [inline server/channels/app/permissions_migrations.go:1109]: Are there other roles we should evaluate for changes here?

**Review by lieut-data — APPROVED** [review]
Wow, nice work @enahum! This was way simpler than I imagined :)

File: server/channels/app/permissions_migrations.go (hunk 1 above)
Comment (enahum) [reply server/channels/app/permissions_migrations.go:1109]: not that I could think of, for all the roles that had `read_channel` this were the only ones that made sense.

Current roles with the `read_channel` permission are:
* System Admin
* Channel User
* Channel Guest
* System Manager
* System read only admin <- I was unsure of this one to be honest
* system user manager

**Review by sbishel — APPROVED** [review]
Nice, this seems like the correct fix for the given infrastructure. Thanks.

**Review by esarafianou — APPROVED** [review]
LGTM, thank you!

**Review by crspeller — APPROVED** [review]

**Comment by enahum** [comment]
Anyone knows why the E2E tests smoketests are failing? should I merge it in this state?

**Comment by mvitale1989** [comment]
Me and Elias paired up to understand why the smoketests were failing, and the root issue is that there is a mismatch between the default roles' permission sets that a clean-state server has, and the defaults that the `Reset to Defaults` button in `System Console -> Permissions -> System Scheme` configures (the smoketests "press" that button during their run, thus were able to catch the inconsistency). In particular, after pressing that button, the `read_channel_content` disappeared from the permission set of any role, leading to the 403s.

FYI this investigation also uncovered that the `get_public_link` permission shard is also missing from the defaults set that `Reset to Defaults` configures, which would require a fix as well.

**Comment by enahum** [comment]
ok this PR is finally passing all tests

**Review by lieut-data — APPROVED** [review]

**Pull request merged**